tests/boardfarm_plugins/boardfarm_prplmesh/tests/prplmesh_base_test.py

```diff
@@ -30,3 +32,143 @@ def prplmesh_status_check(self, entity_or_radio: Union[env.ALEntity, env.Radio])
+
+        Mark failure if no satisfying packet is found.
+
+        Parameters
+        ----------
+        msg: str
+            Message to show in case of failure. It is formatted in a context like
+            "No CMDU <msg> found".
+
+        match_function: Callable[[sniffer.Packet], bool]
+            A function that returns True if it is the expected packet. It is called on every packet
+            returned by get_packet_capture.
+
+        Returns
+        -------
+        [sniffer.Packet]
+            The matching packets.
+        """
+        debug("Checking for CMDU {}".format(msg))
+        result = self.dev.DUT.wired_sniffer.get_cmdu_capture(match_function)
+        assert result, "No CMDU {} found".format(msg)
+        return result
+
+    def check_cmdu_type(
+        self, msg: str, msg_type: int, eth_src: str, eth_dst: str = None, mid: int = None
+    ) -> [sniffer.Packet]:
+        """Verify that the wired sniffer has captured a CMDU.
+
+        Mark failure if the CMDU is not found.
+
+        Parameters
+        ----------
+        msg: str
+            Message to show in case of failure. It is formatted in a context like
+            "No CMDU <msg> found".
+
+        msg_type: int
+            CMDU message type that is expected.
+
+        eth_src: str
+            MAC address of the sender that is expected.
+
+        eth_dst: str
+            MAC address of the destination that is expected. If omitted, the IEEE1905.1 multicast
+            MAC address is used.
+
+        mid: int
+            Message Identifier that is expected. If omitted, the MID is not checked.
+
+        Returns
+        -------
+        [sniffer.Packet]
+            The matching packets.
+        """
+        debug("Checking for CMDU {} (0x{:04x}) from {}".format(msg, msg_type, eth_src))
+        result = self.dev.DUT.wired_sniffer.get_cmdu_capture_type(msg_type, eth_src, eth_dst, mid)
+        assert result, "No CMDU {} found".format(msg)
+        return result
+
+    def check_cmdu_type_single(
+        self, msg: str, msg_type: int, eth_src: str, eth_dst: str = None, mid: int = None
+    ) -> sniffer.Packet:
+        '''Like check_cmdu_type, but also check that only a single CMDU is found.'''
+        cmdus = self.check_cmdu_type(msg, msg_type, eth_src, eth_dst, mid)
+        assert len(cmdus) == 1, "Multiple CMDUs {} found".format(msg)
+        return cmdus[0]
+
+    def check_no_cmdu_type(
+        self, msg: str, msg_type: int, eth_src: str, eth_dst: str = None
+    ) -> NoReturn:
+        '''Like check_cmdu_type, but check that *no* machting CMDU is found.'''
+        debug("Checking for no CMDU {} (0x{:04x}) from {}".format(msg, msg_type, eth_src))
+        result = self.dev.DUT.wired_sniffer.get_cmdu_capture_type(msg_type, eth_src, eth_dst)
+        if result:
+            for packet in result:
+                debug("  {}".format(packet))
+            assert False, "Unexpected CMDU {}".format(msg)
```

Comment:
It's not yet ported from `test_flows.py`. Initial idea was to port methods from from `test_flows.py` when some flow requires it This way we can understand from `git log` why such method was ported.

Reply:
No, check_cmdu_type_single is there. I copy + pasted the exact implementation in test_flows.py. I guess implementation is different because we wanted to be able to print the unexpected packet.

Reply:
As I commented earlier, all cmdu_type methods are copy+pasted from test_flows.py and check_no_cmdu_type() method was like this before this PR. If you don't have any objections I'll resolve this conversation @adam1985d